### ISSUE TITLE
Insecure HTTPS defaults

### ISSUE THREAD

**Issue by voltone** [IanLuites/httpx#2]
Out of the box, HTTPX does not validate the server's certificate:

```elixir
iex(1)> HTTPX.get("https://self-signed.badssl.com/")
{:ok,
 %HTTPX.Response{...}}
```

This is due to the `ssl_options` being passed to Hackney, which are intended to strengthen the security, but end up overriding Hackney's internal defaults such as `:verify_fun` and `:cacertfile`.

**Comment by voltone** [comment]
One solution would be to add `verify: :verify_peer, cacertfile: :certifi.cacertfile()` into the default settings' `ssl_options`(at runtime, not in the `@default_settings`). This would leverage Erlang/OTP's built-in `verify_fun`, which is OK-ish since OTP 20. Some wildcard certificates are only handled correctly in OTP 21 and require adding `customize_hostname_check: [match_fun: :public_key.pkix_verify_hostname_match_fun(:https)]`, which of course breaks older OTP versions.

Using `ssl_verify_fun` the way Hackney does requires extracting the hostname from the URL. The upside is that it performs proper peer certificate and hostname verification also on older OTP versions.

**Comment by IanLuites** [comment]
This will be fixed in release of `v0.1.0`.

The verification survives setting different `ssl_options` and the only way to disable is explicitly disabling the `verify`, `cacertfile`, or `depth`.

The following defaults will always be applied:
```elixir
verify: :verify_peer,
cacertfile: :certifi.cacertfile(),
depth: 2
```

Please re-open if the solution requires additional tweaks.
